### PR TITLE
[Merged by Bors] - chore: tidy various files

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Homology/HomotopyCategory/HomComplex.lean [hunk 1]
```diff
@@ -468,7 +468,7 @@ variable {F G R}
   (δ_hom R F G n m).map_smul k z
 
 @[simp] lemma δ_units_smul (k : Rˣ) (z : Cochain F G n) : δ n m (k • z) = k • δ n m z := by
-  apply δ_smul
+  rw [Units.smul_def, Units.smul_def, δ_smul]
```
Comment (j-loreaux) [inline Mathlib/Algebra/Homology/HomotopyCategory/HomComplex.lean:471]:
```suggestion
@[simp] lemma δ_units_smul (k : Rˣ) (z : Cochain F G n) : δ n m (k • z) = k • δ n m z :=
  δ_smul .. 
```

File: Mathlib/Analysis/Calculus/FDeriv/Analytic.lean [hunk 2]
```diff
@@ -119,7 +119,8 @@ theorem AnalyticOn.iteratedFDeriv [CompleteSpace F] (h : AnalyticOn 𝕜 f s) (n
     -- Porting note: for reasons that I do not understand at all, `?g` cannot be inlined.
     convert ContinuousLinearMap.comp_analyticOn ?g IH.fderiv
     case g => exact ↑(continuousMultilinearCurryLeftEquiv 𝕜 (fun _ : Fin (n + 1) ↦ E) F)
-    rfl
+    simp only [ContinuousLinearMap.strongUniformity_topology_eq, ContinuousLinearEquiv.coe_coe,
+      LinearIsometryEquiv.coe_coe]
```
Comment (j-loreaux) [inline Mathlib/Analysis/Calculus/FDeriv/Analytic.lean:123]: This declaration is a bit slow, but it's slow with or with the `simp only`
```suggestion
    simp
```

File: Mathlib/Analysis/Calculus/FDeriv/Analytic.lean [hunk 3]
```diff
@@ -254,7 +255,8 @@ theorem CPolynomialOn.iteratedFDeriv (h : CPolynomialOn 𝕜 f s) (n : ℕ) :
   · rw [iteratedFDeriv_succ_eq_comp_left]
     convert ContinuousLinearMap.comp_cPolynomialOn ?g IH.fderiv
     case g => exact ↑(continuousMultilinearCurryLeftEquiv 𝕜 (fun _ : Fin (n + 1) ↦ E) F)
-    rfl
+    simp only [ContinuousLinearMap.strongUniformity_topology_eq, ContinuousLinearEquiv.coe_coe,
+      LinearIsometryEquiv.coe_coe]
```
Comment (j-loreaux) [inline Mathlib/Analysis/Calculus/FDeriv/Analytic.lean:259]:
```suggestion
    simp
```

File: Mathlib/Analysis/Calculus/Monotone.lean [hunk 4]
```diff
@@ -191,19 +191,22 @@ theorem Monotone.ae_hasDerivAt {f : ℝ → ℝ} (hf : Monotone f) :
...
+        rw [mem_Ioo] at hy
         rw [mem_Iio]
         norm_num; nlinarith
     -- apply the sandwiching argument, with `g` and the helper function
     apply tendsto_of_tendsto_of_tendsto_of_le_of_le' hx.1 this
     · filter_upwards [self_mem_nhdsWithin]
-      rintro y (hy : y < x)
-      apply div_le_div_of_nonpos_of_le (sub_neg.2 hy).le
+      rintro y hy
+      rw [mem_Iio, ← sub_neg] at hy
```
Comment (j-loreaux) [inline Mathlib/Analysis/Calculus/Monotone.lean:202]: Personally I find the original cleaner. Do you have a reason to prefer the explicit `rw`?

File: Mathlib/Analysis/Calculus/Monotone.lean [hunk 5]
```diff
@@ -191,19 +191,22 @@ theorem Monotone.ae_hasDerivAt {f : ℝ → ℝ} (hf : Monotone f) :
...
+      apply div_le_div_of_nonpos_of_le hy.le
       exact (sub_le_sub_iff_right _).2 (hf.le_rightLim (le_refl _))
     · filter_upwards [self_mem_nhdsWithin]
-      rintro y (hy : y < x)
-      have : ↑0 < (y - x) ^ 2 := sq_pos_of_neg (sub_neg.2 hy)
-      apply div_le_div_of_nonpos_of_le (sub_neg.2 hy).le
+      rintro y hy
+      rw [mem_Iio, ← sub_neg] at hy
```
Comment (j-loreaux) [inline Mathlib/Analysis/Calculus/Monotone.lean:207]: same question

File: Mathlib/GroupTheory/MonoidLocalization.lean [hunk 6]
```diff
@@ -1945,9 +1945,9 @@ theorem leftCancelMulZero_of_le_isLeftRegular
 /-- Given a Localization map `f : M →*₀ N` for a Submonoid `S ⊆ M`,
 if `M` is a cancellative monoid with zero, and all elements of `S` are
 regular, then N is a cancellative monoid with zero.  -/
-theorem isLeftRegular_of_le_IsCancelMulZero (f : LocalizationWithZeroMap S N)
-    [IsCancelMulZero M] (h : ∀ ⦃x⦄, x ∈ S → IsRegular x): IsCancelMulZero N := by
-  have:IsLeftCancelMulZero N:=
+theorem isLeftRegular_of_le_isCancelMulZero (f : LocalizationWithZeroMap S N)
```
Comment (j-loreaux) [inline Mathlib/GroupTheory/MonoidLocalization.lean:1948]: I think the old version is supposed to get a `deprecated` tag now. Please add an inline comment with "deprecated on <current-date>"

File: Mathlib/RingTheory/Localization/Module.lean [hunk 7]
```diff
@@ -95,8 +93,9 @@ variable {M Mₛ : Type*} [AddCommGroup M] [AddCommGroup Mₛ] [Module R M] [Mod
 
 /-- If `M` has an `R`-basis, then localizing `M` at `S` has a basis over `R` localized at `S`. -/
 noncomputable def Basis.ofIsLocalizedModule : Basis ι Rₛ Mₛ :=
-  Basis.mk (b.linearIndependent.of_isLocalizedModule Rₛ S f)
-  (by rw [Set.range_comp, span_eq_top_of_isLocalizedModule Rₛ S _ b.span_eq])
+  Basis.mk
+    (b.linearIndependent.of_isLocalizedModule Rₛ S f)
+    (by rw [Set.range_comp, span_eq_top_of_isLocalizedModule Rₛ S _ b.span_eq])
```
Comment (j-loreaux) [inline Mathlib/RingTheory/Localization/Module.lean:98]:
```suggestion
  .mk (b.linearIndependent.of_isLocalizedModule Rₛ S f) <| by
    rw [Set.range_comp, span_eq_top_of_isLocalizedModule Rₛ S _ b.span_eq]
```

**Review by j-loreaux — COMMENTED** [review]
Thanks!

File: Mathlib/Analysis/Calculus/Monotone.lean (hunk 4 above)
Comment (Ruben-VandeVelde) [reply Mathlib/Analysis/Calculus/Monotone.lean:202]: Mostly just that it relies on the exact definition of the `Ixx`es. `rw` should be robust to changes there (though I suppose those are unlikely), but might also give a better error message if `filter_upwards` ends up creating a different goal here (which may be equally unlikely). Happy to revert these parts if you'd like

**Comment by j-loreaux** [comment]
bors merge

**Comment by j-loreaux** [comment]
Let's try again ...

bors merge

**Pull request closed without merging**